src/api/trace/id_generator.rs

```diff
@@ -0,0 +1,14 @@
+//! # OpenTelemetry Id Generator Interface
+use crate::api;
+use std::fmt;
+
+/// Interface for generating IDs
+pub trait IdGenerator: Send + Sync + fmt::Debug {
+    /// Generate a new `TraceId`
+    fn new_trace_id(&self) -> api::TraceId;
+
+    /// Generate a new `SpanId`
+    fn new_span_id(&self) -> api::SpanId {
+        api::SpanId::from_u64(0)
```

Comment:
Take it or leave it, but IMO a default implementation that would assign every span the ID 0 seems kinda dangerous...

Reply:
Yes thanks for catching that! It was left here by mistake, working on too many things at once...